addons/product/models/product_template.py

```diff
@@ -799,49 +840,39 @@ def _is_combination_possible(self, combination, parent_combination=None):
         """
         self.ensure_one()
 
-        if len(combination) != len(self.valid_product_template_attribute_line_ids):
-            # number of attribute values passed is different than the
-            # configuration of attributes on the template
-            return False
+        if self._can_variant_be_used(combination):
```

Comment:
I still need to check the rest but this makes the diff hard to read.
Just do a preemptive return False here and don't indent the rest.